worker-taskunstowing/src/main/java/com/github/jobservice/workers/taskunstowing/TaskUnstowingWorker.java

```diff
@@ -0,0 +1,195 @@
+        try {
+            stowedTaskRows = databaseClient.getStowedTasks(partitionId, jobId);
+            LOGGER.info("Found {} stowed task(s) for partition id {} and job id {}", stowedTaskRows.size(), partitionId, jobId);
+        } catch (final Exception exception) {
+            final TaskUnstowingWorkerFailure failure = new TaskUnstowingWorkerFailure(
+                TaskUnstowingWorkerFailure.FAILED_TO_READ_FROM_DATABASE_ID,
+                String.format("Failed to read stowed task(s) for partition id %s and job id %s from database", partitionId, jobId));
+            processFailure(failure, exception, document);
+            return;
+        }
+
+        for (final StowedTaskRow stowedTaskRow : stowedTaskRows) {
+            final TaskMessage taskMessage;
+            try {
+                taskMessage = convertStowedTaskRowToTaskMessage(stowedTaskRow);
+            } catch (final IOException exception) {
+                final TaskUnstowingWorkerFailure failure = new TaskUnstowingWorkerFailure(
+                    TaskUnstowingWorkerFailure.FAILED_TO_CONVERT_DATABASE_ROW_TO_TASK_MESSAGE_ID,
+                    String.format("Failed to convert database row with partition id {}, job id {} and job task id {} to task message",
+                                  stowedTaskRow.getPartitionId(), stowedTaskRow.getJobId(), stowedTaskRow.getTrackingInfoJobTaskId()));
+                processFailure(failure, exception, document);
+                continue;
+            }
+
+            try {
+                workerTaskData.sendMessage(taskMessage);
+                LOGGER.info("Sent unstowed task message with partition id {}, job id {} and job task id {} to queue {}",
+                            partitionId, jobId, taskMessage.getTracking().getJobTaskId(), taskMessage.getTo());
+                try {
+                    databaseClient.deleteStowedTask(partitionId, jobId);
+                } catch (final Exception exception) {
+                    final TaskUnstowingWorkerFailure failure = new TaskUnstowingWorkerFailure(
+                        TaskUnstowingWorkerFailure.FAILED_TO_DELETE_UNSTOWED_TASK_MESSAGE_FROM_DATABASE_ID,
+                        String.format("Sent unstowed task message with partition id %s, job id %s and job task id %s to queue %s, but "
+                            + "failed to delete unstowed task message from database", partitionId, jobId,
+                                      taskMessage.getTracking().getJobTaskId(), taskMessage.getTo()));
+                    if (DatabaseExceptionChecker.isTransientException(exception)) {
+                        LOGGER.error(failure.getFailureMsg(), exception);
+                        throw new DocumentWorkerTransientException(failure.getFailureMsg(), exception);
+                    } else {
+                        processFailure(failure, exception, document);
+                    }
+                }
+            } catch (final Exception exception) {
```

Comment:
What exception is this trying to catch here?  As is it this defeats the transient issue handling above by catching the `DocumentWorkerTransientException` and turning it into a non-transient failure.  I suspect the catch block shouldn't be present at all.

Reply:
I'm catching the exception that might be thrown when trying to send a message to rabbit, but its in the wrong place, I've moved it now.